### PR TITLE
Filter duplicate messages to avoid noise in loggly closes #2315

### LINKED ISSUE #2315: Filter duplicate Loggly messages
We don't ideally want the client to flood Loggly with multiple copies of the same message.

### PR DESCRIPTION
**Pull request by myleshorton**

### CODE REVIEW

**Comment by myleshorton** [comment]
You want to take a look at this one @fffw?

File: src/github.com/getlantern/flashlight/logging/logging.go [hunk 1]
```diff
@@ -37,7 +39,9 @@ var (
 	errorOut io.Writer
 	debugOut io.Writer
 
-	lastAddr string
+	lastAddr   string
+	duplicates = make(map[string]bool)
+	dupLock    = &sync.Mutex{}
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/logging/logging.go]: `sync.Mutex` doesn't require initialization

File: src/github.com/getlantern/flashlight/logging/logging.go [hunk 2]
```diff
@@ -161,7 +165,30 @@ type logglyErrorWriter struct {
...
+	}
+
+	return false
+}
+
 func (w logglyErrorWriter) Write(b []byte) (int, error) {
+	fullMessage := string(b)
+	if isDuplicate(fullMessage) {
+		msg := fmt.Sprintf("Not logging duplicate: %v", fullMessage)
+		fmt.Println(msg)
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/logging/logging.go]: We can still use `log.Tracef` or `log.Debugf ` here.

File: src/github.com/getlantern/flashlight/logging/logging.go [hunk 3]
```diff
@@ -161,7 +165,30 @@ type logglyErrorWriter struct {
...
+		return 0, errors.New(msg)
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/logging/logging.go]: Is duplication considered an error?

**Comment by fffw** [comment]
Thanks for this quick workaround @myleshorton that I should have done. It's good in general apart from those somewhat nitpicking comments above.

File: src/github.com/getlantern/flashlight/logging/logging.go (hunk 1 above)
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/logging/logging.go]: Oh right -- we can just use its zero value. Makes sense.

**Comment by myleshorton** [comment]
OK cool thanks @fffw -- just made those changes!

**Comment by fffw** [comment]
It works as expected. LGTM!

**Pull request merged**